urlc175f05cb8_mchllbrt_PermCode_tgz-pJ8-permlib_examples_UniversalSeparableJ8/src/permlab/examples/LCW.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * To change this template, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package permlab.examples;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import permlib.Permutation;
+
+/**
+ *
+ * @author MichaelAlbert
+ */
+public class LCW {
+
+    /**
+     * @param args the command line arguments
+     */
+    public static void main(String[] args) {
+        
+        Permutation p = new Permutation("5 8 4 6 1 3 2 9 7");
+        ArrayList<Integer> pa = new ArrayList<Integer>();
+        for (int i = 0; i < p.length(); i++) {
+            pa.add(i);
+        }
+        HashSet<ArrayList<Integer>> rectangles = new HashSet<ArrayList<Integer>>();
+        rectangles.add(pa);
+        System.out.println(doCheck(p, rectangles, 2));
+        HashSet<ArrayList<Integer>> splits = splits(p, 6, pa);
+        for(ArrayList<Integer> a : splits) System.out.println(a);
+        
+//        final HashSet<Permutation> basis = new HashSet<Permutation>();
+//        final int k = 2;
+//        for (int n = 7; n <= 10; n++) {
+//            PermClass c = new PermClass(basis);
+//            c.processPerms(n,
+//                    new PermProcessor() {
+//                        @Override
+//                        public void process(Permutation p) {
+//                            ArrayList<Integer> pa = new ArrayList<Integer>();
+//                            for (int i = 0; i < p.length(); i++) {
+//                                pa.add(i);
+//                            }
+//                            HashSet<ArrayList<Integer>> rectangles = new HashSet<ArrayList<Integer>>();
+//                            rectangles.add(pa);
+//                            if (!doCheck(p, rectangles, k)) {
+//                                System.out.println(p);
+//                                basis.add(p);
+//                            };
+//                        }
+//
+//                        @Override
+//                        public void reset() {
+//                            throw new UnsupportedOperationException("Not supported yet.");
+//                        }
+//
+//                        @Override
+//                        public String report() {
+//                            throw new UnsupportedOperationException("Not supported yet.");
+//                        }
+//                    });
+//        }
+    }
+
+    public static HashSet<ArrayList<Integer>> splits(Permutation p, int index, Collection<ArrayList<Integer>> rectangles) {
+        HashSet<ArrayList<Integer>> result = new HashSet<ArrayList<Integer>>();
+        for (ArrayList<Integer> rectangle : rectangles) {
+            result.addAll(splits(p, index, rectangle));
+        }
+        return result;
+    }
+
+    public static HashSet<ArrayList<Integer>> splits(Permutation p, int index, ArrayList<Integer> rectangle) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.